### PR TITLE
[DOCS] Adds inference phase to how a DFA job works section

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds the `inference` phase and its description to the `How a data frame analytics job works` section. Applies to 7.9 and above.

### Preview

[How a DFA job works – Inference phase](https://stack-docs_1327.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-dfa-phases.html#ml-dfa-phases-inference)

### CODE REVIEW

File: docs/en/stack/ml/df-analytics/ml-dfa-phases.asciidoc [hunk 1]
```diff
@@ -56,8 +58,17 @@ Only the additional fields that the analysis calculated are written back, the
 ones that have been loaded in the loading data phase are not. The 
 {dfanalytics-job} matches the results with the data rows in the destination 
 index, merges them, and indexes them back to the destination index. When the 
-process is complete, the task is marked as completed and the {dfanalytics-job} 
-stops. Your data is ready to be evaluated.
+process is complete in case of an {oldetection} job, the task is marked as
```
Comment (dimitris-athanasiou) [inline docs/en/stack/ml/df-analytics/ml-dfa-phases.asciidoc:61]: I wonder if the last sentence (`the task is marked as completed...`) should move into the end of all the phases so we don't have to explain it twice. We could have a paragraph at the end that goes along the lines `Finally, after all phases are completed the task is marked as completed...`.

File: docs/en/stack/ml/df-analytics/ml-dfa-phases.asciidoc [hunk 2]
```diff
@@ -56,8 +58,17 @@ Only the additional fields that the analysis calculated are written back, the
...
-process is complete, the task is marked as completed and the {dfanalytics-job} 
-stops. Your data is ready to be evaluated.
+process is complete in case of an {oldetection} job, the task is marked as 
+completed and the {dfanalytics-job} stops. Your data is ready to be evaluated.
+
+[[ml-dfa-phases-inference]]
+== {infer-cap}
+
+This phase exists only for {regression} and {classification} jobs. In this 
+phase, the job infers the test split of the data set agains the trained model.
```
Comment (dimitris-athanasiou) [inline docs/en/stack/ml/df-analytics/ml-dfa-phases.asciidoc:68]: typo: `agains` -> `against`

**Comment by szabosteve** [comment]
@dimitris-athanasiou Thanks for your suggestions. I addressed them via 93fa065.

**Comment by szabosteve** [comment]
**Backport commits:**
7.x: 8636e36
7.9: 4d592a8

**Pull request merged**